src/main/drivers/barometer/barometer_lps.c

```diff
@@ -0,0 +1,166 @@
+#define LPS25HB_NO_MA               0x00
+#define LPS25HB_MA_2                0x01
+#define LPS25HB_MA_4                0x03
+#define LPS25HB_MA_8                0x07
+#define LPS25HB_MA_16               0x0F
+#define LPS25HB_MA_32               0x1F
+
+bool lpsWriteCommand(busDevice_t *busdev, uint8_t cmd, uint8_t byte) {
+    return spiBusWriteRegister(busdev, cmd, byte);
+}
+
+bool lpsReadCommand(busDevice_t *busdev, uint8_t cmd, uint8_t len, uint8_t *data) {
+    return spiBusReadRegisterBuffer(busdev, cmd | 0x80 | 0x40, data, len);
+}
+
+static void lpsOn(busDevice_t *busdev, uint8_t CTRL1_val) {
+    lpsWriteCommand(busdev, LPS25HB_CTRL1, CTRL1_val | 0x80);
+    delay(500);
+}
+
+static void lpsOff(busDevice_t *busdev) {
+    lpsWriteCommand(busdev, LPS25HB_CTRL1, 0x00 | (0x01 << 2));
+}
+
+uint32_t rawP = 0;
+uint32_t rawT = 0;
+
+static void lps_nothing(baroDev_t *baro) {
+	UNUSED(baro);
+	return;
+}
+
+static void lps_read(baroDev_t *baro) {
+    uint8_t status = 0x00;
+    lpsReadCommand(&baro->busdev, LPS25HB_STATUS, 1, &status);
+    if (status & 0x03) {
+    		uint8_t temp[5];
+    		lpsReadCommand(&baro->busdev, LPS25HB_OUT_XL, 5, temp);
+
+    		/* Build the raw data */
+    		for(uint8_t i=0; i<3; i++)
+    			rawP |= (((uint32_t)temp[i]) << (8*i));
+    		if (rawP & 0x00800000)
+    		rawP |= 0xFF000000;
+
+    		//Temp in C
+    		uint16_t temperature_tmp = ((uint16_t) temp[4] << 8) | (uint16_t) temp[3];
+    		rawT = (((uint16_t)temperature_tmp / 48 + 425) * 10);
+    } else {
+    		rawP = 0;
+    		rawT = 0;
+    }
+}
+
+static void lps_calculate(int32_t *pressure, int32_t *temperature) {
+	*pressure = (int32_t)((float)((int32_t)rawP) * 6.25f); //should be divided by 40.96f
+	*temperature = (int32_t)rawT;
+}
+
+bool lpsDetect(baroDev_t *baro)
+{
+
+    delay(10); // No idea how long the chip takes to power-up, but let's make it 10ms
+
+//Detect
+    busDevice_t *busdev = &baro->busdev;
+    IOInit(busdev->busdev_u.spi.csnPin, OWNER_BARO_CS, 0);
+    IOConfigGPIO(busdev->busdev_u.spi.csnPin, IOCFG_OUT_PP);
+    IOHi(busdev->busdev_u.spi.csnPin); // Disable
+    spiSetDivisor(busdev->busdev_u.spi.instance, SPI_CLOCK_STANDARD); // XXX
+
+    uint8_t temp = 0x00;
+    lpsReadCommand(&baro->busdev, LPS25HB_WHO_AM_I, 1, &temp);
+    if (temp != LPS25HB_ID) {
+        goto fail;
+    }
+
+//Init
+    lpsOff(busdev);
+    lpsWriteCommand(busdev, LPS25HB_CTRL2, (0x00 << 1));
+    lpsWriteCommand(busdev, LPS25HB_RES_CONF, (LPS25HB_AVT_64 | LPS25HB_AVP_512));
+    lpsWriteCommand(busdev, LPS25HB_CTRL4, 0x01);
+    lpsOn(busdev, (0x04 << 4) | (0x01 << 1) | (0x01 << 2) | (0x01 << 3));
+
+    lpsReadCommand(busdev, LPS25HB_CTRL1, 1, &temp);
+
+    baro->ut_delay = 1;
+    baro->up_delay = 1000000 / 24;
+    baro->start_ut = lps_nothing;
+    baro->get_ut = lps_nothing;
+    baro->start_up = lps_nothing;
+    baro->get_up = lps_read;
+    baro->calculate = lps_calculate;
+    lps_read(baro);
+    delay(50);
```

Comment:
Why this delay?

Reply:
Delays come from my own experience with this chip as I had trouble initializing it to correct values otherwise. Also after init, I had to usually wait a bit for data. No idea why it is I will do more testing on it...

Reply:
This may cause nasty bugs in future - it's better to fix underlying problem